### PR TITLE
Adding non-nominal tilts

### PR DESCRIPTION
**Pull request by charlesrp**

### CODE REVIEW

File: pysiaf/tests/test_nirspec.py [hunk 1]
```diff
@@ -40,10 +40,12 @@ def test_against_test_data():
     test_data_dir = os.path.join(TEST_DATA_ROOT, instrument)
 
 
-    include_tilt = False
+    include_tilt = True
```
Comment (Johannes-Sahlmann) [inline pysiaf/tests/test_nirspec.py]: Please modify the test such that both cases are executed in a loop (code below will have to be indented of course):
`for include_tilt in [False, True]:`

File: pysiaf/tests/test_nirspec.py [hunk 2]
```diff
@@ -149,6 +155,13 @@ def test_nirspec_aperture_transforms(verbose=False):
 
         # aperture
         aperture = siaf[aper_name]
+        # offset slightly from default tilt values
```
Comment (Johannes-Sahlmann) [inline pysiaf/tests/test_nirspec.py]: Do I understand correctly that this is to add the tilt case to the tests? 
If yes, I suggest to add another loop in there which makes sure that both `tilt=None` and `tilt=something` are tested.
In the current implementation it seems like only the `tilt=something` case is exercised.

File: pysiaf/tests/test_nirspec.py [hunk 3]
```diff
@@ -189,7 +202,8 @@ def test_nirspec_slit_transformations(verbose=False):
     """
     siaf = Siaf(instrument)
 
-    threshold = 0.1
+    threshold = 0.012  # 12 mas
+    pixel_threshold = 10 * threshold
```
Comment (Johannes-Sahlmann) [inline pysiaf/tests/test_nirspec.py]: The new thresholds used in the assert statements below seem to be much more loose than before. Why did that become necessary?

**Comment by charlesrp** [comment]
Hello @Johannes-Sahlmann 

I made most of the changes you requested and pushed the modified version to my gwa-tilt branch.
Regarding the error thresholds in test_nirspec_slit_transformations(), my thought here is that since NIRSpec pixels are about 0.1", if we want to test for 0.01" precision, the threshold in pixel units should then be about 0.1. Some of the current discrepancies in the tests are as big as 0.08 pixels (independent of whether tilt is None or not), so the revised thresholds do let the test pass. What precision we should really be aiming for in this test should probably be a separate discussion.

Charles

**Pull request merged**